explainaboard/processors/argument_pair_extraction.py

```diff
@@ -0,0 +1,273 @@
+            "n_reply_tokens": feature.Value(
+                dtype="float",
+                description="the number of review sentence",
+                func=lambda info, x, c: c.block_reply_tokens,
+            ),
+            "n_reply_position": feature.Value(
+                dtype="float",
+                description="the number of review sentence",
+                func=lambda info, x, c: c.block_reply_position,
+            ),
+        }
+
+        return [
+            AnalysisLevel(
+                name='example',
+                features=features,
+                metric_configs=self.default_metrics(),
+            ),
+            AnalysisLevel(
+                name='block',
+                features=block_features,
+                metric_configs=self.default_metrics(level='block'),
+            ),
+        ]
+
+    def default_analyses(self) -> list[Analysis]:
+        analysis_levels = self.default_analysis_levels()
+        block_features = analysis_levels[1].features
+        analyses: list[Analysis] = [
+            BucketAnalysis(
+                level="block",
+                description=block_features["true_label"].description,
+                feature="block_true_label",
+                method="discrete",
+                number=8,
+            )
+        ]
+
+        analyses.extend(self.continuous_feature_analyses())
+        return analyses
+
+    def _get_true_label(self, data_point):
+        """
+        Get the true label from a data point. Overloaded from parent class.
+        :param data_point: the data point under consideration
+        :return: the true label for the output
+        """
+        return data_point["true_tags"]
+
+    def _get_predicted_label(self, data_point):
+        """
+        Get the predicted label from a data point. Overloaded from parent class.
+        :param data_point: the data point under consideration
+        :return: the predicted label for the output
+        """
+        return data_point["pred_tags"]
+
+    @aggregating()
+    def _statistics_func(self, samples: Iterator, sys_info: SysOutputInfo):
+        vocab, vocab_rank = accumulate_vocab_from_samples(
+            samples,
+            lambda x: " ".join(x['sentences']),
+            unwrap(sys_info.source_tokenizer),
+        )
+
+        return {'vocab': vocab, 'vocab_rank': vocab_rank}
+
+    def _gen_cases_and_stats(
+        self,
+        sys_info: SysOutputInfo,
+        sys_output: list[dict],
+        statistics: Any,
+        analysis_level: AnalysisLevel,
+    ) -> tuple[list[AnalysisCase], list[MetricStats]]:
+        if analysis_level.name == 'example':
+            return super()._gen_cases_and_stats(
+                sys_info, sys_output, statistics, analysis_level
+            )
+        elif analysis_level.name != 'block':
+            raise ValueError(f'{analysis_level.name}-level analysis not supported')
+        # Do span-level analysis
+        cases: list[AnalysisCaseLabeledBlock] = []
```

Comment:
It will following errors if we don't cast
```shell
xplainaboard/processors/argument_pair_extraction.py:228: error: "AnalysisCase" has no attribute "true_label"
explainaboard/processors/argument_pair_extraction.py:229: error: "AnalysisCase" has no attribute "predicted_label"
```

Reply:
Oh I see, it makes sense. It would be great if some comments are put around this.

Reply:
Sure and done.